src/internal/core_actor.cc

```diff
@@ -245,8 +245,11 @@ caf::behavior core_actor_state::make_behavior() {
       std::vector<peer_info> result;
       for (const auto& [peer_id, state] : peers) {
         endpoint_info info{peer_id, state->addr()};
-        result.emplace_back(peer_info{std::move(info), peer_flags::remote,
-                                      peer_status::connected});
+        auto status = peer_statuses->get(peer_id);
+        if (status != peer_status::unknown) {
```

Comment:
You likely know better, but wondering a bit why it's okay to filter `unknown` here? There seems to be a peer in peers, but we decide to not return its information.

Reply:
The `peer_statuses` is a shared map. If we get `unknown`, it basically means that the element has been removed in the meantime. I would need to look at the sources to see whether the connector (background task for making the handshaking) can remove items or if we do that only in the core actor at the moment.

But details aside, it's a thing that other threads may modify while we iterate over the list of locally known peers and `unknown` just means "no longer exists". 🙂

Reply:
Oh, and after reading `unknown` here, we would eventually receive an event that cleans up the state in `peers` in case your wondering.